activesupport/lib/active_support/cache/strategy/local_cache.rb

```diff
@@ -75,7 +75,10 @@ def delete_entry(key, options)
           end
 
           def fetch_entry(key, options = nil) # :nodoc:
-            @data.fetch(key) { @data[key] = yield }
+            entry = @data.fetch(key) { @data[key] = yield }
+            dup_entry = entry.dup
```

Comment:
Why you need to dup the entry itself?

Reply:
the `@data` hash keeps a reference to the entry, if I don't dup it, user can modify the underlying value which would affect the cache.